src/main/java/com/splunk/logging/HttpEventCollectorEventInfo.java

```diff
@@ -21,17 +21,17 @@
 /**
  * Container for Splunk http event collector event data
  */
-public class HttpEventCollectorEventInfo {
+public class HttpEventCollectorEventInfo<E> {
```

Comment:
Got it. <T> is more standard for a single generic type, so that through me off. The idea of supporting JSON objects is definitely something we're supportive of.

The current suggestion of a generic type / making the event a string OR an object is inconsistent with how we've addressed this in our other logging libs like .NET. We'd like to keep as much consistent as possible.

To be consistent with that design, HttpEventCollectorEventInfo and HttpEventCollectorSender would not be generic types.

Then a "Data" property of type object would be added to HttpEventCollectorSenderInfo as in [here] (https://github.com/splunk/splunk-library-dotnetlogging/blob/master/src/Splunk.Logging.Common/HttpEventCollectorEventInfo.cs#L97)

This would not be a breaking change with the existing infra and would then allow data objects to be sent. Can you could update your code to match that design?




1. In its current state I